DS9 8.6 recipe updates to build on osx-64 and link against conda-forge X11 stack

jehturner opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->
Hi @epassaro,

I've had a go at merging my work on DS9 8.6 (from Astroconda) into your contemporaneous conda-forge feedstock. Sorry this has taken so long to get back to! This principally gets the package building on MacOS for Intel (the same binary can also run fine on ARM, if you use the Intel distribution there) and produces a Linux version that's linked against the conda-forge X11 stack (with scalable fonts) instead of the system's (with bitmap fonts). As discussed previously, these are both X11 builds.

This is my first conda-forge submission (though I'm quite au fait with conda packaging & conda-forge in general); I'll leave it as a WIP until we see what happens with the CI...

A few details to consider:
* I've included the associated `xpa` utilities here (some of our users might expect to continue having them).
* I'm aware that v8.7 should be out soon, but it took significant effort to get 8.6 working fully and I think it would be good to get this build upstream in the meantime. The next version might be harder, since it includes a major Tk update, apparently. Whichever of us gets to that first, the patches here might be useful for reference.
* Currently, we're still using the Intel conda distribution on Apple ARM machines, but I hope to work on native ARM builds at some point (there are a few potential complications that I've been investigating a bit with the Google chatbot, to get a better idea where we are!).
* I've added my username as a maintainer; is that appropriate?

Thanks!

James.

jehturner committed:
Merge my changes/patches from Astroconda ds9 into the new conda-forge recipe.
  Patch upstream source to build for X11 on MacOS (using conda-forge libs).
  Link correctly against the conda-forge X11 stack (+ scalable fonts) on Linux.
Allow for libxml2-devel being split out from libxml2 on conda-forge.
Include complete dependencies.
Reorganize requirements slightly to eliminate various conda-build warnings.
Include associated xpa utilities in the conda package.

jehturner committed:
Make sure we don't try to build for Windows, which probably won't work.

jehturner committed:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.03.20.15.24

Other tools:
- conda-build 25.9.0
- rattler-build 0.52.0
- rattler-build-conda-compat 1.4.10

jehturner committed:
Apply linting suggestions from Google AI / tidy up:

  Undo splitting of run-time deps into a separate output section, as the warning this eliminates is reportedly harmless and the single `outputs` pattern has been known to lead to run_exports breakage.
  Restore make dependency, for consistency.
  Invoke test command in a way that won't break with no DISPLAY.
  Add recommended license_family metadata.

  Remove old LDFLAGS settings (which could break future cross-compilation etc.).
  Fix code signing on MacOS (TBC), in preparation for an ARM64 build.
  Build in parallel.
  Quote strings containing variables, to be safe.

jehturner committed:
Remove autoconf dep, which should ordinarily be unnecessary.
Sign xpa binaries as well as ds9 and add a clarifying comment.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

epassaro commented:
Hi James, nice to hear from you. I can review this in a couple of days, maybe more

jehturner committed:
Revert libxml2-devel to older libxml2, until conda-forge pinning gets updated.

jehturner committed:
Restore autotools, since some bundled libs try to regenerate their scripts on conda-forge (time-stamp issues?).
Touch files we have patched, to protect them from being overwritten by this.

jehturner committed:
Add dependencies still needed by the older libxml2 on conda-forge.
Don't ignore_run_exports for libxml2, in spite of conda-build warning, because the in-memory internals of ds9 are actually linked against it.

jehturner committed:
Need to use liblzma-devel rather than liblzma, to get the soname link, in spite of patching out the source dependency in libtiff.

jehturner committed:
Restore build dep on "zip", which isn't in conda-forge's Docker container.

jehturner commented:
Hi Ezequiel,

Sorry -- my changes since opening this PR have mostly been putting a few things back how you had them :see_no_evil:. That's because `conda-forge` is pinning `libxml2` to an older version than I was picking up locally and also to account for a couple of things that are missing/different in the Docker container, as you've obviously discovered previously... I should have my bearings with conda-forge maintenance after this!

Cheers,

James.

PS. I can probably squash some of these commits if they seem too noisy, though some of the associated explanation might be good to keep...

jehturner commented:
Finally green! I was looking for the build artifacts, to give them a quick test, and it appears that one has to enable `azure: store_build_artifacts: True` in `conda-forge.yml` in order to get them; does that sound good?

https://conda-forge.org/docs/maintainer/conda_forge_yml/#azure

epassaro commented:
So the build artifacts are no longer uploaded automatically? No problem.

I can review this during the weekend, It looks very good. So there's nothing to review besides the patches to the source code and the recipe, right?  (the rest of the files seems to be modified by conda-smithy re-rendering).

EDIT: also tell me if you are done with this PR and I can squash-merge the PR.

jehturner commented:
I think the packages will still get uploaded if you merge the PR, but they don't get saved in the meantime, unless you change the `conda-forge.yml` setting. I was just going to give them a quick check before they get published, but they should be OK now. From a better look at the build logs, I'm pretty sure they are linked correctly.

The only other thing that's not generated automatically is `conda_build_config.yaml` (which I believe is pretty standard). This targets (eg.) CentOS 7+ and MacOS 10.13+, which are the same minimum versions that we and Anaconda / conda-forge have been defaulting to (as I understand it), at least until recently. The latest Miniforge requirements are consistent with this:

https://github.com/conda-forge/miniforge?tab=readme-ov-file#requirements-and-installers

Thanks!

James.

epassaro commented:
> I think the packages will still get uploaded if you merge the PR, but they don't get saved in the meantime, unless you change the `conda-forge.yml` setting. I was just going to give them a quick check before they get published, but they should be OK now. From a better look at the build logs, I'm pretty sure they are linked correctly.
> 

Yes, that exactly what I thought. They are trying to save some space, probably. We should activate that when debugging stuff in a PR, no need to merge that.

> The only other thing that's not generated automatically is `conda_build_config.yaml` (which I believe is pretty standard). This targets (eg.) CentOS 7+ and MacOS 10.13+, which are the same minimum versions that we and Anaconda / conda-forge have been defaulting to (as I understand it), at least until recently. The latest Miniforge requirements are consistent with this:
> 
> https://github.com/conda-forge/miniforge?tab=readme-ov-file#requirements-and-installers
> 
> Thanks!
> 
> James.

Ok, if you want I can merge it now.

jehturner commented:
I've just realized that `conda-smithy` would make a few more changes to the CI version pins after my last `meta.yaml` edits. After a bit of head scratching, I believe that should not break anything (if done now).

jehturner commented:
@conda-forge-admin, please rerender :face_with_peeking_eye:

Commit:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.11.16.09.5

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ Recipe maintainer "jehturner" does not exist


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/20143257523](https://github.com/conda-forge/conda-forge-webservices/actions/runs/20143257523). Examine the logs at this URL for more detail.</sub>

jehturner commented:
~Apparently I would need to create an "Anaconda Cloud" account to resolve that new error (not 100% sure why it just started showing up). I'll have a look at that;~ do you want to remove the offending line for now, or add me as a maintainer? Obviously from Gemini / NOIR Lab's point of view, we'd like to be able to get `ds9` builds released when needed if we're going to keep them upstream, but it would not be my intention to start changing things around without co-ordination :wink:. But it doesn't matter too much, as long as we can get updates merged in a reasonable amount of time.

jehturner committed:
Remove new maintainer, to resolve linter error until added to conda-forge org. [skip ci]

jehturner commented:
Never mind; it seems that the Anaconda Cloud thing was just an AI search mode hallucination! I've just removed the offending line for now, because it looks like I would need adding to the conda-forge organization first. So I think this is ready (again). And I think the failure showed up because the admin re-render does more checks than my local one.

jehturner commented:
Ack, I just realized that my username is still in various places in the CI scripts after pushing that last change, so I probably need to do @conda-forge-admin, please rerender again...

Commit:
MNT: Re-rendered with conda-smithy 3.53.3 and conda-forge-pinning 2025.12.12.23.31.0

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

epassaro commented:
not sure what needs to be done here, but I will add you as a manteiner no matter what. Just ping me when you need me to merge this one

jehturner commented:
It's done now, thanks!
